stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -354,6 +354,49 @@ function copyto!(A::AbstractSparseMatrixCSC, B::AbstractSparseMatrixCSC)
     return A
 end
 
+copyto!(A::AbstractMatrix, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+# Ambiguity resolution
+copyto!(A::PermutedDimsArray, B::AbstractSparseMatrixCSC) = _sparse_copyto!(A, B)
+
+function _sparse_copyto!(dest::AbstractMatrix, src::AbstractSparseMatrixCSC{T}) where {T}
+    dest === src && return dest
+    checkbounds(dest, axes(src)...)
+    fill!(dest, zero(T))
+    @inbounds for col in 1:size(src, 2), ptr in nzrange(src, col)
+        row = rowvals(src)[ptr]
+        val = nonzeros(src)[ptr]
+        dest[row, col] = val
+    end
+    return dest
+end
+
+function copyto!(dest::AbstractMatrix{T}, Rdest::CartesianIndices{2},
+                 src::AbstractSparseMatrixCSC, Rsrc::CartesianIndices{2}) where {T}
+    isempty(Rdest) && return dest
+    if size(Rdest) != size(Rsrc)
+        throw(ArgumentError("source and destination must have same size (got $(size(Rsrc)) and $(size(Rdest)))"))
+    end
+    checkbounds(dest, first(Rdest))
+    checkbounds(dest, last(Rdest))
+    checkbounds(src, first(Rsrc))
+    checkbounds(src, last(Rsrc))
+    src′ = Base.unalias(dest, src)
+    for I in Rdest
+        @inbounds dest[I] = zero(T)
+    end
+    rows, cols = Rsrc.indices
+    lin = LinearIndices(Base.IdentityUnitRange.(Rsrc.indices))
```

Comment:
I'm not familiar with the specifics of `OffsetArrays`, for which, I believe, this line is. I suggest that we have a test for a case when this line is nontrivial.

Reply:
This line is defined so that `lin[row,col]` gives the linear index of `(row, col)` in `Rsrc`. Thus, each `(row, col)` in `Rsrc` gets copied to `(row´, col´) = Tuple(Rdest[lin[row, col]])`. This is already tested by the new `@test A[Rdest] == A´[Rdest] == Matrix(B)[Rsrc]`